### PR TITLE
dts: msm8909: add acer-t01 and mobvoi-catfish

### PR DESCRIPTION
**Pull request by wonderfulShrineMaidenOfParadise**
-next seems to be preferred.
Since old lk2nd is still useful to boot rootfs, I would keep this PR until -next gets released.

- Redo accidentally deleted #243
- A fork of #155

Cc: @TipzRickyCheung @CE1CECL

### CODE REVIEW

**Comment by TipzRickyCheung** [comment]
I will try it out with a few of my msm8909 and report back.

**Comment by CE1CECL** [comment]
If you have catfish @TipzRickyCheung I had one working for the new branch with display: https://github.com/ChrisEric1/lk2nd

**Comment by TipzRickyCheung** [comment]
> If you have catfish @TipzRickyCheung I had one working for the new branch with display: https://github.com/ChrisEric1/lk2nd

I do, will give it a go later. My port didn't actually work properly in that there was no display for any of those devices.

**Comment by CE1CECL** [comment]
> > If you have catfish @TipzRickyCheung I had one working for the new branch with display: https://github.com/ChrisEric1/lk2nd
> 
> I do, will give it a go later. My port didn't actually work properly in that there was no display for any of those devices.

Just note the power button works but you cant use multifunc to move the menu/enter it

**Comment by TipzRickyCheung** [comment]
> > > If you have catfish @TipzRickyCheung I had one working for the new branch with display: https://github.com/ChrisEric1/lk2nd
> > 
> > 
> > I do, will give it a go later. My port didn't actually work properly in that there was no display for any of those devices.
> 
> Just note the power button works but you cant use multifunc to move the menu/enter it

Did you not try to map it with `lk2nd,keys` in dts? You can refer to keylayouts in Android or kernel source for key codes.

Here's an example:
```
lk2nd,keys =
	<KEY_VOLUMEDOWN 107 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>,
	<KEY_VOLUMEUP   108 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
```

**Comment by CE1CECL** [comment]
> > > > If you have catfish @TipzRickyCheung I had one working for the new branch with display: https://github.com/ChrisEric1/lk2nd
> > > 
> > > 
> > > I do, will give it a go later. My port didn't actually work properly in that there was no display for any of those devices.
> > 
> > 
> > Just note the power button works but you cant use multifunc to move the menu/enter it
> 
> Did you not try to map it with `lk2nd,keys` in dts? You can refer to keylayouts in Android or kernel source for key codes.
> 
> Here's an example:
> 
> ```
> lk2nd,keys =
> 	<KEY_VOLUMEDOWN 107 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>,
> 	<KEY_VOLUMEUP   108 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
> ```

Actually not yet, when I last tried lk2nd, the pmOS I made worked with stock kernel, but was built out of pmbootstrap. Mainline may work but untested (no dsi was there at the time)

**Comment by TipzRickyCheung** [comment]
> > > > > If you have catfish @TipzRickyCheung I had one working for the new branch with display: https://github.com/ChrisEric1/lk2nd
> > > > 
> > > > 
> > > > I do, will give it a go later. My port didn't actually work properly in that there was no display for any of those devices.
> > > 
> > > 
> > > Just note the power button works but you cant use multifunc to move the menu/enter it
> > 
> > 
> > Did you not try to map it with `lk2nd,keys` in dts? You can refer to keylayouts in Android or kernel source for key codes.
> > Here's an example:
> > ```
> > lk2nd,keys =
> > 	<KEY_VOLUMEDOWN 107 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>,
> > 	<KEY_VOLUMEUP   108 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
> > ```
> 
> Actually not yet, when I last tried lk2nd, the pmOS I made worked with stock kernel, but was built out of pmbootstrap. Mainline may work but untested (no dsi was there at the time)

So I am assuming mainline development is picking up pace? Good to hear any development anyways since catfish is slow on stock. I was actually attempting to port LineageOS for it, but never got the graphical stack working properly.

Maybe I should be refrained from talking about Android development, but catfish would of course be a great addition to the pmOS family. Unfortunately can't be a part of it cos I have stepped into the time where I should focus on academics.

**Comment by CE1CECL** [comment]
> > > > > > If you have catfish @TipzRickyCheung I had one working for the new branch with display: https://github.com/ChrisEric1/lk2nd
> > > > > 
> > > > > 
> > > > > I do, will give it a go later. My port didn't actually work properly in that there was no display for any of those devices.
> > > > 
> > > > 
> > > > Just note the power button works but you cant use multifunc to move the menu/enter it
> > > 
> > > 
> > > Did you not try to map it with `lk2nd,keys` in dts? You can refer to keylayouts in Android or kernel source for key codes.
> > > Here's an example:
> > > ```
> > > lk2nd,keys =
> > > 	<KEY_VOLUMEDOWN 107 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>,
> > > 	<KEY_VOLUMEUP   108 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
> > > ```
> > 
> > 
> > Actually not yet, when I last tried lk2nd, the pmOS I made worked with stock kernel, but was built out of pmbootstrap. Mainline may work but untested (no dsi was there at the time)
> 
> So I am assuming mainline development is picking up pace? Good to hear any development anyways since catfish is slow on stock. I was actually attempting to port LineageOS for it, but never got the graphical stack working properly.
> 
> Maybe I should be refrained from talking about Android development, but catfish would of course be a great addition to the pmOS family. Unfortunately can't be a part of it cos I have stepped into the time where I should focus on academics.

On lineage, i got stuck at bootanim using the GSI, but I had to disable SELinux for it to show up

**Comment by TipzRickyCheung** [comment]
> 

A GSI works?

**Comment by CE1CECL** [comment]
> > 
> 
> A GSI works?

I had to `cp -nrfv /stock.system.img/* /gsi.system.img/ && setfacl --restore ...` on it but it got stuck at splash screen

**Comment by wonderfulShrineMaidenOfParadise** [comment]
@TipzRickyCheung @CE1CECL do you still use Matrix? [#msm8916:postmarketos.org](https://app.element.io/#/room/#msm8916:postmarketos.org)
I have some ideas about the panel.

File: dts/msm8909/apq8009w-wtp.dts
Comment (TipzRickyCheung) [inline dts/msm8909/apq8009w-wtp.dts:1]: Fix that spacing before it's too late.

File: dts/msm8909/msm8909-1gb-qrd-skuc.dts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+#include <skeleton.dtsi>
+
+/ {
+	// This is used by the bootloader to find the correct DTB
+	qcom,msm-id = <245 0>;
+	qcom,board-id = <0x1000b 0x8>, <0x11000b 0x8>, <0x21000b 0x8>;
+
+	model = "Qualcomm Technologies, Inc. MSM8909 QRD SKUC";
+	compatible = "qcom,msm8909-qrd", "qcom,msm8909", "qcom,qrd", "lk2nd,device";
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8909/msm8909-1gb-qrd-skuc.dts:13]: Not sure if `FIXME` is preferred.

File: dts/msm8909/apq8009w-wtp.dts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+#include <skeleton.dtsi>
+
+/ {
+	// This is used by the bootloader to find the correct DTB
+	qcom,msm-id = <265 0>, <301 0>;
+	qcom,board-id= <8 0x105>;
+
+	model = "Qualcomm Technologies, Inc. APQ8009W-PM8916 MTP";
+	compatible = "qcom,apq8009-mtp", "qcom,apq8009", "qcom,mtp";
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8909/apq8009w-wtp.dts:13]: I am confused. `MTP` or `WTP`?
https://android.googlesource.com/kernel/msm/+/android-msm-mullet-3.18-oreo-wear-dr/arch/arm/boot/dts/qcom/apq8009w-wtp.dts#24
https://android.googlesource.com/kernel/msm/+/android-msm-mullet-3.18-oreo-wear-dr/arch/arm/boot/dts/qcom/apq8009w-1gb-wtp.dts#24

Comment (TipzRickyCheung) [reply dts/msm8909/apq8009w-wtp.dts:13]: WTP is a thing, but I think you wouldn't want to do this as they use custom board IDs.

Comment (wonderfulShrineMaidenOfParadise) [reply dts/msm8909/apq8009w-wtp.dts:13]: `qcom,board-id= <8 0x105>;` fits `apq8009w-wtp.dts`. I would assume it's shared by several devices.
For board IDs, do you have downstream dts for some references?

Comment (wonderfulShrineMaidenOfParadise) [reply dts/msm8909/apq8009w-wtp.dts:13]: I see https://github.com/Nexusma/msm-catfish-9.0-wear-os/blob/master/arch/arm/boot/dts/qcom/catfish.dts#L20-L21

However, `CDP` is supposed to be `<8 0x103>`
https://android.googlesource.com/kernel/msm/+/android-msm-mullet-3.18-oreo-wear-dr/arch/arm/boot/dts/qcom/apq8009w-cdp.dts#27

File: README.md [hunk 3]
```diff
@@ -47,7 +47,7 @@ page on the EFIDroid wiki for an exact mapping of LK targets to SoCs.
 ## Supported devices
 ### lk2nd-msm8909
 - Acer Liquid Z330 - T01
-- Mobvoi TicWatch Pro - catfish
+- Mobvoi TicWatch Pro - catfish (use `lk2nd-msm8909-appended-dtb.img`)
```
Comment (CE1CECL) [inline README.md:50]: Only a comment but this should work on all of the following codenames: `catfish, catshark, catfish_ext` with lk2nd-msm8909-appended-dtb.img (WIP for lk1st by me and @wonderfulShrineMaidenOfParadise in IRC) 

NOTE: EDL cable is not optional but REQUIRED, as well as the micro-usb to TicWatch adapter that usually saus "Watch Pro" on the back (not my typo), else you will be responsible.

File: README.md [hunk 4]
```diff
@@ -36,7 +36,7 @@ https://source.codeaurora.org/quic/la/kernel/lk/) (tag `LA.BR.1.2.9.1-02310-8x16
 which is a heavily modified version of the [Little Kernel Embedded Operating System].
 
 ## Supported SoCs
-- `lk2nd-msm8909`: APQ8009, MSM8909
+- `lk2nd-msm8909`: APQ8009W, MSM8909
```
[the remark this answers is not in the document]
Comment (wonderfulShrineMaidenOfParadise) [reply README.md:39]: Perhaps, if we have more devices with `APQ8009` or `MSM8909W`?

File: README.md (hunk 3 above)
Comment (wonderfulShrineMaidenOfParadise) [reply README.md:50]: Current dts: https://github.com/Nexusma/msm-catfish-9.0-wear-os/blob/master/arch/arm/boot/dts/qcom/catfish.dts

I cannot promise for catshark, as the dts for it is not added here:
https://github.com/Nexusma/msm-catfish-9.0-wear-os/blob/master/arch/arm/boot/dts/qcom/catshark.dts

**Pull request closed without merging**